src/pages/scan/Scan.tsx

```diff
@@ -84,9 +87,28 @@ export default function ScanPage() {
     setHasPermission(false);
   };
 
+  const onFileUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (!file) {
+      return;
+    }
+
+    try {
+      const decodedText = await scanFile(file);
+      onScanResult(decodedText, null);
+    } catch (e: any) {
+      errorModal({title: 'Error processing QR code', content: e.message});
+    }
+  };
+
   return (
     <div>
       <TopNav backBtnText="Scan" backNavigateTo={-1} />
+      {!processing && process.env.NODE_ENV === 'development' && (
```

Comment:
Would it make sense to always show it in development, but also in production in case the app is viewed on a desktop browser instead of a mobile device?

Alternatively I'd also be fine w/ some way to override it (putting something in manually in localStorage, a setting etc.) ;)

Reply:
> Would it make sense to always show it in development, but also in production in case the app is viewed on a desktop browser instead of a mobile device

added ;)